exercises/practice/circular-buffer/CircularBuffer.ps1

```diff
@@ -0,0 +1,72 @@
+<#
+.SYNOPSIS
+Implement the circular buffer data structure.
+
+.DESCRIPTION
+A circular buffer, cyclic buffer or ring buffer is a data structure that uses a single, fixed-size buffer as if it were connected end-to-end.
+Please implement the circular buffer class with these methods:
+- Write     : write new value into the buffer, raise error if the buffer is full.
+- Overwrite : overwrite the oldest element in the buffer if the buffer is full, otherwise behave like write.
+- Clear     : clear all elements in the buffer, it is now empty.
+- Read      : read the oldest element in the buffer, and return its value.
+
+There is a predefined ToString method to help debugging and print out the buffer just like in instruction.
+
+.EXAMPLE
+$buffer = [CircularBuffer]::new(5)
+
+$buffer.ToString()
+Return: [ ][ ][ ][ ][ ]
+
+1..5 | Foreach-Object {$buffer.Write($_)}
+$buffer.ToString()
+Return: [1][2][3][4][5]
+
+$buffer.Read()
+Return: 1
+
+$buffer.ToString()
+Return: [ ][2][3][4][5]
+
+$buffer.Write('A')
+$buffer.ToString()
+Return: [A][2][3][4][5]
+
+$buffer.Overwrite('B')
+$buffer.ToString()
+Return: [A][B][3][4][5]
+#>
+
+Class CircularBuffer {
+    $buffer
+
+    CircularBuffer() {
+        Throw "Please implement this class"
+    }
+
+    Write() {
+        Throw "Please implement this function"
+    }
+
+    Overwrite() {
+        Throw "Please implement this function"
+    }
+
+    Clear() {
+        Throw "Please implement this function"
+    }
+
+    Read() {
+        Throw "Please implement this function"
+    }
+
+    [string] ToString() {
```

Comment:
When using the online editor will there be no tests for this function.

Reply:
I don't think this is meant as something the student implements, but just as a helper function

Reply:
@glaxxie ?

Reply:
I've removed the tostring method and the optional test in the stub files to avoid confusing students. It stays in the example file. I do hope the tostring for the binary search tree exercise will stay though.